pkg/kubelet/server/stats/types.go

```diff
@@ -139,11 +147,45 @@ type VolumeStats struct {
 // FsStats contains data about filesystem usage.
 type FsStats struct {
 	// AvailableBytes represents the storage space available (bytes) for the filesystem.
-	AvailableBytes *resource.Quantity `json:"availableBytes,omitempty"`
+	AvailableBytes *uint64 `json:"availableBytes,omitempty"`
 	// CapacityBytes represents the total capacity (bytes) of the filesystems underlying storage.
-	CapacityBytes *resource.Quantity `json:"capacityBytes,omitempty"`
+	CapacityBytes *uint64 `json:"capacityBytes,omitempty"`
 	// UsedBytes represents the bytes used for a specific task on the filesystem.
 	// This may differ from the total bytes used on the filesystem and may not equal CapacityBytes - AvailableBytes.
 	// e.g. For ContainerStats.Rootfs this is the bytes used by the container rootfs on the filesystem.
-	UsedBytes *resource.Quantity `json:"usedBytes,omitempty"`
+	UsedBytes *uint64 `json:"usedBytes,omitempty"`
+}
+
+// UserDefinedMetricType defines how the metric should be interpreted by the user.
+type UserDefinedMetricType string
+
+const (
+	// Instantaneous value. May increase or decrease.
+	MetricGauge UserDefinedMetricType = "gauge"
+
+	// A counter-like value that is only expected to increase.
+	MetricCumulative UserDefinedMetricType = "cumulative"
+
+	// Rate over a time period.
+	MetricDelta UserDefinedMetricType = "delta"
+)
+
+// UserDefinedMetricDescriptor contains metadata that describes a user defined metric.
+type UserDefinedMetricDescriptor struct {
+	// The name of the metric.
+	Name string `json:"name"`
+
+	// Type of the metric.
+	Type UserDefinedMetricType `json:"type"`
+
+	// Display Units for the stats.
+	Units string `json:"units"`
+}
+
+// UserDefinedMetric represents a metric defined and generate by users.
+type UserDefinedMetric struct {
+	UserDefinedMetricDescriptor `json:",inline"`
+	// Value of the metric. Float64s have 53 bit precision.
+	// We do not forsee any metrics exceeding that value.
+	Value float64 `json:"value"`
```

Comment:
FYI: @timstclair: This field is not optional. Just highlighting it in case you missed it.